src/psi/j/executors/flux.py

```diff
@@ -0,0 +1,278 @@
+    _state_map = {'NEW': JobState.QUEUED,
+                  'DEPEND': None,
+                  'SCHED': None,
+                  'RUN': JobState.ACTIVE,
+                  'CLEANUP': None,
+                  'INACTIVE': JobState.COMPLETED}
+    _event_map = {'submit': JobState.QUEUED,
+                  # 'alloc': None,
+                  'start': JobState.ACTIVE,
+                  'finish': JobState.COMPLETED,
+                  # 'release': None,
+                  # 'free': None,
+                  # 'clean': None,
+                  'exception': JobState.FAILED}
+
+    def __init__(self, url: Optional[str] = None,
+                 config: Optional[JobExecutorConfig] = None) -> None:
+        """
+        Initializes a `FluxJobExecutor`.
+
+        :param url: Not used, but required by the spec for automatic initialization.
+        :param config: The `FluxJobExecutor` does not have any configuration options.
+        :type config: psi.j.JobExecutorConfig
+        """
+        # TODO: url is not passed
+        # if not url.startswith('flux://'):
+        #     raise ValueError('expected `flux://` url')
+
+        super().__init__(url=url, config=config)
+
+        self._jobs = dict()  # {job.uid: [job, fut, flux_id]}
+        self._idmap = dict()  # {flux_id, job.uid}
+        self._lock = ru.RLock('flux_adaptor')  # lock state updates
+
+        self._fh = self._ru.FluxHelper()
+        self._finfo = self._fh.start_service()
+        self._fuid = self._finfo['uid']
+        self._fex = self._fh.get_executor(self._fuid)
+        self._fh = self._fh.get_handle(self._fuid)
+
+    # # TODO: how to free resources?
+    # def close(self) -> None:
+    #     try:
+    #         if self._fh and self._fuid:
+    #             self._fh.close_service(self._fuid)
+    #     except Exception:
+    #         pass
+
+    def _jobid_cb(self, fut: Any, jpsi_id: str) -> None:
+
+        jpsi_job, _ = self._jobs[jpsi_id]
+        flux_id = fut.jobid()
+
+        # NOTE: the check below seems to kick in sometimes, but apparently the
+        #       cb is then invoked again and succeeds - we seem to get all job
+        #       ids eventually...
+        #
+        # From James Corbett: You can rely on the jobid callback always firing
+        #       first. As Flux's bindings are now, FluxExecutor has to get the
+        #       jobid before it can even request event updates, and the executor
+        #       triggers the jobid callbacks before requesting event updates.
+        #       [...] I don't think it would change unless there were fairly
+        #       radical changes to the underlying Flux functions.
+
+        if flux_id == -1:
+            logger.warn('invalid flux id for %s', jpsi_id)
+            return
+
+        logger.debug('register flux jobid for %s: %s', jpsi_id, flux_id)
+
+        self._idmap[flux_id] = jpsi_id
+
+        jpsi_job._native_id = flux_id
+        job_status = JobStatus(JobState.QUEUED, time=time.time())
```

Comment:
I am not sure.  I guess we should do that consistently across all connectors then.  But since the native ID is now a first class property on the `Job`, I don't think that is needed anymore?